pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,373 @@
+// +build windows
+
+package ebpf
+
+/*
+//! These includes are needed to use constants defined in the ddfilterapi
+#include <WinDef.h>
+#include <WinIoCtl.h>
+
+//! Defines the objects used to communicate with the driver as well as its control codes
+#include "c/ddfilterapi.h"
+*/
+import "C"
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"golang.org/x/sys/windows"
+)
+
+// HandleType expresses what of DriverHandle one is - implicitly implies if filters have been placed on the driver for the handle
+type HandleType string
+
+const (
+	driverFile = `\\.\ddfilter`
+
+	// FlowHandle is keyed to return 5-tuples from the driver that represents a flow. Used with: (#define FILTER_LAYER_TRANSPORT ((uint64_t) 1)
+	FlowHandle HandleType = "Flow"
+
+	// DataHandle is keyed to return full packets from the driver. Used with: #define FILTER_LAYER_IPPACKET ((uint64_t) 0)
+	DataHandle HandleType = "Data"
+
+	// StatsHandle has no filter set and is used to pull total stats from the driver
+	StatsHandle HandleType = "Stats"
+)
+
+var (
+	// Buffer holding datadog driver filterapi (ddfilterapi) signature to ensure consistency with driver.
+	ddAPIVersionBuf = makeDDAPIVersionBuffer(C.DD_FILTER_SIGNATURE)
+)
+
+// Creates a buffer that Driver will use to verify proper versions are communicating
+// We create a buffer because the system calls we make need a *byte which is not
+// possible with const value
+func makeDDAPIVersionBuffer(signature uint64) []byte {
+	buf := make([]byte, C.sizeof_uint64_t)
+	binary.LittleEndian.PutUint64(buf, signature)
+	return buf
+}
+
+// DriverInterface holds all necessary information for interacting with the windows driver
+type DriverInterface struct {
+	driverFlowHandle  *DriverHandle
+	driverStatsHandle *DriverHandle
+
+	path       string
+	totalFlows int64
+}
+
+// NewDriverInterface returns a DriverInterface struct for interacting with the driver
+func NewDriverInterface() (*DriverInterface, error) {
```

Comment:
feedback on DriverInterface in general: is there a distinction between methods called createX vs generateX  ? If not, can we standardize on one (I suggest `createX`)

Reply:
The only distinction that I had (and the original reason I had this different) was that the generate method was used with the DriverInterface, while the create methods were used with the DriverHandle structs. 

If this is just additional confusion/unnecessary - happy to change everything to just be "create".